yesod-bin/Devel.hs

```diff
@@ -170,27 +174,28 @@ reverseProxy opts appPortVar = do
                 manager
         defaultSettings' = maybe id (setHost . fromString) (develHost opts) defaultSettings
         runProxyTls port app = do
-          let cert = $(embedFile "certificate.pem")
-              key = $(embedFile "key.pem")
-              tlsSettings = tlsSettingsMemory cert key
-          runTLS tlsSettings (setPort port defaultSettings') $ \req send -> do
-            let req' = req
-                    { requestHeaders
-                        = ("X-Forwarded-Proto", "https")
-                        -- Workaround for
-                        -- https://github.com/yesodweb/wai/issues/478, where
-                        -- the Host headers aren't set. Without this, generated
-                        -- URLs from guestApproot are incorrect, see:
-                        -- https://github.com/yesodweb/yesod-scaffold/issues/114
-                        : (case lookup "host" (requestHeaders req) of
-                            Nothing ->
-                                case requestHeaderHost req of
-                                    Just host -> (("Host", host):)
-                                    Nothing -> id
-                            Just _ -> id)
-                          (requestHeaders req)
-                    }
-            app req' send
+            let certDef = $(embedFile "certificate.pem")
+                keyDef = $(embedFile "key.pem")
+                certOpts = bisequence $ (certPath &&& keyPath) opts
```

Comment:
I'm personally not a fan of this style of code, I'd prefer an explicit `case` here for readability. But I think if you follow my recommendation below the point will be moot.